compiler/rustc_lint/src/passes.rs

```diff
@@ -127,6 +127,7 @@ macro_rules! declare_combined_late_lint_pass {
         }
 
         impl $name {
+            #[allow(redundant_field_initializers)]
```

Comment:
Shouldn't we prevent this by checking if the struct expression span is from expansion?

Reply:
Yeah, already fixed.